addon/resolve-attribute-util.js

```diff
@@ -114,13 +116,32 @@ export function resolveValue(key, value, modelName, store, schema, record, paren
   const schemaInterface = recordData.schemaInterface;
 
   // First check to see if given value is either a reference or an array of references
-  let reference = computeAttributeReference(key, value, modelName, schemaInterface, schema);
-  if (reference !== undefined && reference !== null) {
+  let computedValue, nested, reference;
+  if (schema.useComputeAttribute()) {
+    computedValue = computeAttribute(key, value, modelName, schemaInterface, schema);
+    if (computedValue && typeof computedValue === 'object') {
```

Comment:
should confirm that this is the best way to check for POJOness in a hot path

Reply:
good catch, I asked and it turned out that checking for null explicitly is better, fixed